### PR TITLE
Document deriveBits in the EcdhPrivateKey Class

### PR DESCRIPTION
**Pull request by HamdaanAliQuatil**

### CODE REVIEW

**Comment by HamdaanAliQuatil** [comment]
Split from #114 
The `@template` for the `@macro` used is defined in #127

File: lib/src/webcrypto/webcrypto.ecdh.dart [hunk 1]
```diff
@@ -38,6 +38,19 @@ abstract class EcdhPrivateKey {
     return impl.ecdhPrivateKey_generateKey(curve);
   }
 
+  /// Derives a shared secret from two ECDH key pairs using the private key from one pair 
+  /// and the public key from another.
```
Comment (jonasfj) [inline lib/src/webcrypto/webcrypto.ecdh.dart:42]:
```suggestion
  /// Derive a shared secret from two ECDH key pairs using the private key from one pair 
  /// and the public key from another.
```

**Review by jonasfj — APPROVED** [review]

File: lib/src/webcrypto/webcrypto.ecdh.dart [hunk 2]
```diff
@@ -38,6 +38,19 @@ abstract class EcdhPrivateKey {
...
+  /// 
+  /// The shared secret is identical whether using A's private key and B's public key, 
+  /// or B's private key and A's public key, enabling secure key exchange between the 
+  /// two parties.
+  ///
+  /// [length] specifies the length of the derived secret in bits.
```
Comment (jonasfj) [inline lib/src/webcrypto/webcrypto.ecdh.dart:48]: Should we make a list bullet point list with what lengths the different curves can provide:
```md
[length] specifies the length of the derived secret in bits. The maximum length depends on the [EllipticCurve] used in the keys:
...
 * [EllipticCurve.p521] can provide up to 528 bits.
```

or something like that.

Comment (jonasfj) [reply lib/src/webcrypto/webcrypto.ecdh.dart:48]: We'd have to find out how much each curve can derive, and make sure we have tests :D

Comment (jonasfj) [reply lib/src/webcrypto/webcrypto.ecdh.dart:48]: Actually, let's do this in a follow up issue.. That way we land now.

Landing now is always better :rofl:

Comment (jonasfj) [reply lib/src/webcrypto/webcrypto.ecdh.dart:48]: Filed https://github.com/google/webcrypto.dart/issues/130

**Pull request merged**